module/zfs/dmu_send.c

```diff
@@ -1682,6 +1689,8 @@ issue_data_read(struct send_reader_thread_arg *srta, struct send_range *range)
 	if (arc_err != 0) {
 		srdp->abd = abd_alloc_linear(srdp->datasz, B_FALSE);
 		srdp->io_outstanding = B_TRUE;
+		srdp->io_compressed = zioflags & ZIO_FLAG_RAW ||
+		    zioflags & ZIO_FLAG_RAW_COMPRESS;
```

Comment:
```
#define	ZIO_FLAG_RAW	(ZIO_FLAG_RAW_COMPRESS | ZIO_FLAG_RAW_ENCRYPT)
```
So we only need to check one of RAW or RAW_COMPRESS.  I'm guessing RAW_COMPRESS would be enough?

Reply:
I wasn't sure if zioflags fits in 32 bits, so it might be worth explicitly converting this to a boolean_t so that we don't accidentally truncate the high bits.  i.e. `(zioflags & ZIO_FLAG_WRITE_COMPRESS) != 0`.